examples/flutter_gallery/lib/demo/cupertino/cupertino_navigation_demo.dart

```diff
@@ -0,0 +1,740 @@
+                        ),
+                      ),
+                      const Padding(padding: const EdgeInsets.only(top: 20.0)),
+                      new Row(
+                        mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                        children: <Widget>[
+                          new CupertinoButton(
+                            color: CupertinoColors.activeBlue,
+                            minSize: 30.0,
+                            padding: const EdgeInsets.symmetric(horizontal: 24.0),
+                            borderRadius: new BorderRadius.circular(32.0),
+                            child: const Text(
+                              'GET',
+                              style: const TextStyle(
+                                fontSize: 14.0,
+                                fontWeight: FontWeight.w700,
+                                letterSpacing: -0.28,
+                              ),
+                            ),
+                            onPressed: () {},
+                          ),
+                          new CupertinoButton(
+                            color: CupertinoColors.activeBlue,
+                            minSize: 30.0,
+                            padding: EdgeInsets.zero,
+                            borderRadius: new BorderRadius.circular(32.0),
+                            child: const Icon(CupertinoIcons.ellipsis, color: CupertinoColors.white),
+                            onPressed: () {},
+                          ),
+                        ],
+                      ),
+                    ],
+                  ),
+                ),
+              ],
+            ),
+          ),
+          const Padding(
+            padding: const EdgeInsets.only(left: 16.0, top: 28.0, bottom: 8.0),
+            child: const Text(
+              'USERS ALSO LIKED',
+              style: const TextStyle(
+                color: const Color(0xFF646464),
+                letterSpacing: -0.60,
+                fontSize: 15.0,
+                fontWeight: FontWeight.w500,
+              ),
+            ),
+          ),
+          new SizedBox(
+            height: 200.0,
+            child: new ListView.builder(
+              scrollDirection: Axis.horizontal,
+              itemCount: 10,
+              itemExtent: 160.0,
+              itemBuilder: (BuildContext context, int index) {
+                return new Padding(
+                  padding: const EdgeInsets.only(left: 16.0),
+                  child: new Container(
+                    decoration: new BoxDecoration(
+                      borderRadius: new BorderRadius.circular(8.0),
+                      color: new Color.fromARGB(
+                        255,
+                        (color.red + random.nextInt(100) - 50).clamp(0, 255),
+                        (color.green + random.nextInt(100) - 50).clamp(0, 255),
+                        (color.blue + random.nextInt(100) - 50).clamp(0, 255),
```

Comment:
this will cause the colour to change every time you build, in an unpredictable manner. I recommend instead creating a local math.Random, seeded from `index`, so that the colour will be consistent every time a row is built.

Reply:
Thanks.

Moved the generation to the top of the whole demo. Moved the generation of the inner page to a State.